modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -932,6 +983,42 @@ bool OculusModule::updateModule()
 
         if (buttonMapping > 0)
         {
+            if (m_useVive)
+            {
+                if (!m_frameTransformInterface->frameExists(m_headFrameName))
+                {
+                    yError() << "[OculusModule::updateModule] The frame named " << m_rootFrameName
+                             << " does not exist.";
+                    yError() << "[OculusModule::updateModule] I will not start the walking. Please "
+                                "try to start again.";
+                    return true;
+                }
+
+                yarp::sig::Matrix oculusInertial_T_oculusRoot = identitySE3();
+                if (!m_frameTransformInterface->getTransform(
+                        m_headFrameName, m_rootFrameName, oculusInertial_T_oculusRoot))
+                {
+                    yError() << "[OculusModule::updateModule] Unable to evaluate the "
+                             << m_headFrameName << " to " << m_rootFrameName << "transformation";
+                    yError() << "[OculusModule::updateModule] I will not start the walking. Please "
+                                "try to start again.";
+                    return true;
+                }
+
+                // get only the yaw axis
+                iDynTree::Rotation tempRot(getRotation(oculusInertial_T_oculusRoot));
+                double yaw = 0;
+                double dummy = 0;
+                tempRot.getRPY(dummy, dummy, yaw);
```

Comment:
Note that here you are assuming that the Z axis is parallel to gravity, but according to https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#reference-spaces it is the Y axis pointing upward

Reply:
`oculusInertial_T_oculusRoot` arrives from the transform server so it should contain a rotation matrix then we get only the yaw (idyntree definition). I think this does not depends on the name of the frame of openxr.